bundle/src/main/java/com/adobe/acs/commons/adobeio/core/service/impl/EndpointServiceImpl.java

```diff
@@ -0,0 +1,397 @@
+import static org.apache.sling.api.servlets.HttpConstants.METHOD_POST;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.CharEncoding;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.time.StopWatch;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPatch;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.metatype.annotations.Designate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.adobe.acs.commons.adobeio.core.config.EndpointConfiguration;
+import com.adobe.acs.commons.adobeio.core.service.EndpointService;
+import com.adobe.acs.commons.adobeio.core.service.HttpClientService;
+import com.adobe.acs.commons.adobeio.core.service.IntegrationService;
+import com.adobe.acs.commons.adobeio.core.types.Filter;
+import com.adobe.acs.commons.adobeio.core.types.PKey;
+import com.adobe.acs.commons.adobeio.exception.AdobeIOException;
+import com.drew.lang.annotations.NotNull;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+@SuppressWarnings("PackageAccessibility")
+@Component(
+        service = EndpointService.class,
+        immediate = true,
+        property = {
+                Constants.SERVICE_DESCRIPTION + "=Adobe I/O. Endpoint",
+                Constants.SERVICE_VENDOR + "=Adobe I/O",
+                "webconsole.configurationFactory.nameHint" + "=Endpoint"}
+)
+@Designate(ocd = EndpointConfiguration.class, factory = true)
+public class EndpointServiceImpl implements EndpointService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EndpointServiceImpl.class);
+    private EndpointConfiguration config;
+    
+    @Reference
+    private IntegrationService integrationService;
+
+    @Reference
+    private HttpClientService httpClientService;
+
+    private String endpointId;
+
+    @Activate
+    @Modified
+    protected void activate(final EndpointConfiguration config) throws AdobeIOException {
+        LOGGER.debug("Start ACTIVATE Endpoint {}", config.getId());
+        this.config = config;
+        this.endpointId = config.getId();
+        LOGGER.debug("End ACTIVATE Endpoint {}", endpointId);
+
+        if (null == this.integrationService) {
+            throw new AdobeIOException("Integration-service not defined");
```

Comment:
unnecessary -- if integrationService isn't injected, the component won't be activated.